### PR TITLE
Blog project 001

### PR DESCRIPTION
**Pull request by Josphat205**
### **Procedures**

- Created a new project
- Initialized using git
- Added Postgresql as the database
- Created migrations
- Added foreign key

### CODE REVIEW

File: db/migrate/20221122095639_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :text
+      t.references :user, null: false, foreign_key: true
+
+      t.timestamps
+    end
+  end
+end
```
Comment (ShadyShawkat) [inline db/migrate/20221122095639_create_posts.rb:11]: https://github.com/Josphat205/rails-blog/blob/e49d14bb26f5baccd82a3191b618f2b384c99b8d/db/migrate/20221122094053_create_users.rb#L1-L11

- [x] Kindly add a `posts_counter` column to the `Users` table, as per the following [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).

File: db/migrate/20221122095639_create_posts.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :text
+      t.references :user, null: false, foreign_key: true
+
```
Comment (ShadyShawkat) [inline db/migrate/20221122095639_create_posts.rb:7]: - [x] Please rename the `user` foreign key column to `author` instead, as per the following [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).

File: db/migrate/20221122101411_create_comments.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.text :text
+      t.references :user, null: false, foreign_key: true
```
Comment (ShadyShawkat) [inline db/migrate/20221122101411_create_comments.rb:5]: - [x] Please rename the `user` foreign key column to `author` instead, as per the following [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).

File: db/migrate/20221122102306_create_likes.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :user, null: false, foreign_key: true
```
Comment (ShadyShawkat) [inline db/migrate/20221122102306_create_likes.rb:4]: - [x] Please rename the `user` foreign key column to `author` instead, as per the following [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @Josphat205, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20221122095639_create_posts.rb (hunk 1 above)
Comment (Josphat205) [reply db/migrate/20221122095639_create_posts.rb:11]: The posts counter is already exist, i added using a different migration

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- 

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**